site/assets/scripts/app.js

```diff
@@ -77,11 +86,11 @@ $(document).ready(function() {
   
   $("#geo_layers").change(function(e) {
     const layer = $(e.target).val()
-    const geopromise = loadGeoLayer(layer);
+    const geopromise = loadGeoLayer(layer, DataCatalog);
     App.map.removeLayer(App.selected_layer);
 
     // temp hack to force CCD data to load:
-    if (App.dataSet.valueSet.geometry != App.geoLayer.geometry) {
+    if (App.dataSet.valueSet.geometry != App.geoLayer.geometry || datasourceChanged) {
```

Comment:
rather than having a global datasourceChanged boolean can you pass dataSourceChanged as an extra parameter on trigger?

see:

http://api.jquery.com/trigger/